### PR TITLE
Remove feature is implemented

### LINKED ISSUE #3: Implement the Remove feature
Implement the use case for the remove action.

### PR DESCRIPTION
**Pull request by quiz3**
You don't have to accept, as this feature is still under construction, but I thought you might want to see what I've been working on

### CODE REVIEW

**Comment by quiz3** [comment]
Remove feature has been implemented.

Still pending:
- switch from default values and ifs to match

File: src/actions/remove.rs [hunk 1]
```diff
@@ -1,7 +1,90 @@
+use crate::actions::remove;
 use crate::args::RemoveArgs;
 use crate::config::UserConfig;
+use std::io::{stdout, Write, stdin};
+use mprs::utils;
+use mprs::utils::{print_table, list_dir};
+use std::fs::remove_file;
+use std::io;
+use std::path::Path;
+use prettytable::{Table, Cell, Row};  // should this be in utils?
```
Comment (gursi26) [inline src/actions/remove.rs]: Yes it should

File: src/actions/remove.rs [hunk 2]
```diff
@@ -1,7 +1,90 @@
...
+    println!("Playlist path: {:?}", selected_playlist);
+
+    // Using utils, get path of all songs in playlist and store in a vector.
+    let paths = list_dir(&selected_playlist.to_owned());
+
+    // Declare index representing which song to delete (assigned via user input if no query term is given).
+    let id_idx: i32;
+
+    // Print all songs in playlist and let user decide which to remove.
+    if query == "None" {
```
Comment (gursi26) [inline src/actions/remove.rs]: Match statements are the idiomatic way to do this in rust. Please use those.

File: src/actions/remove.rs (hunk 1 above)
Comment (quiz3) [reply src/actions/remove.rs]: Yeah, I had these imported 'cause I thought I'd have to build the table myself. Turns out I figured out how to use `utils::print_table` with what I had.

Comment (quiz3) [reply src/actions/remove.rs]: So remove just line 10, right?

File: src/actions/remove.rs (hunk 2 above)
Comment (quiz3) [reply src/actions/remove.rs]: Yes sir. I will do this another day, however.

**Review by gursi26 — CHANGES_REQUESTED** [review]
Overall, code does not work. Deletes the wrong song, no table shows up and if the song is not found it just crashes. Please fix.

**Comment by quiz3** [comment]
> Overall, code does not work. Deletes the wrong song, no table shows up and if the song is not found it just crashes. Please fix.

That is not what happens on my machine at all.

Here is what happens:
IMG_1511

**Comment by quiz3** [comment]
> Overall, code does not work. Deletes the wrong song, no table shows up and if the song is not found it just crashes. Please fix.

Just to clarify, the feature to be implemented has the following specifications:
- If playlist is specified but song is not specified, list all the songs in the playlist and offer decision to user.
- If playlist is not specified and song is specified, search for the song in _every_ playlist.
- If neither playlist nor song is specified, **TO BE DETERMINED**
- If both playlist and song are specified, search for song in playlist directory. 

In any case, if a song name is provided and there are no matches, print error message and then offer alternative options to the user.

If a song name is provided and there are multiple matches, print them in a table and offer the user a choice between them.

**Comment by gursi26** [comment]
That is correct. To add on;

- If neither playlist nor song is specified, print all songs across all folders and let the user choose (the user would not typically use this, but in case they do).
- Also, if the provided playlist is not found, an error message should be printed.
- When a query term is provided and there is only 1 match, delete that song, whenever there is more than 1 term that matches the query, display in a table and let the user decide.

Also, `RemoveArgs` in `args.rs` can be modified to reflect the fact that playlists are optional for this command and that the `-q` flag is no longer required for a query (refer to `PlayArgs` for how to do this).

**Comment by quiz3** [comment]
The remove feature is now fully implemented.

**Comment by quiz3** [comment]
> Overall, code does not work. Deletes the wrong song, no table shows up and if the song is not found it just crashes. Please fix.

It works now.

**Comment by quiz3** [comment]
I added a playlist column to the table (sometimes) printed during the process of the remove use case. 

I did this because I discovered through testing that a reasonable user may want to put the same song in two different playlists. To allow the user to then modify those playlists by removing songs, it becomes necessary to differentiate between playlists, as the song would otherwise appear identically multiple times in the table of options.

**Comment by gursi26** [comment]
Looks very noice. Tested it out and it works great.

**Review by gursi26 — APPROVED** [review]

**Pull request merged**